Objects/regions.c

```diff
@@ -1954,3 +1955,39 @@ void _PyRegion_set_cown_parent(PyObject* bridge, PyObject* cown) {
     Py_XINCREF(cown);
     Py_XSETREF(data->cown, cown);
 }
+
+void _Py_RegionRemoveReference(PyObject *src, PyObject *tgt) {
+    if (Py_REGION(src) == Py_REGION(tgt)) {
+        // Nothing to do -- intra-region references have no accounting.
+        return;
+    }
+
+    // If the target is local, then so must the source be. So this should
+    // be covered by the previous check.
+    assert(!_Py_IsLocal(tgt));
+
+    if (_Py_IsImmutable(tgt) || _Py_IsCown(tgt)) {
+        // Nothing to do -- removing a ref to an immutable or a cown has no additional accounting.
+        return;
+    }
+
+    regionmetadata* tgt_md = Py_REGION_DATA(tgt);
+    if (_Py_IsLocal(src)) {
+        // Dec LRC of the previously referenced region
+        // TODO should this decrement be a function, if it hits zero,
+        // then a region could become unreachable.
+        tgt_md->lrc -= 1;
```

Comment:
Modifying the LRC be its own function to do the open and close operations if needed propagate the status as well to any parent region or cown.

Reply:
I've started working on this in my branch